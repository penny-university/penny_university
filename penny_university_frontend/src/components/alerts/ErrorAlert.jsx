import React, { useState } from 'react'
import { Alert } from 'reactstrap'
import PropTypes from 'prop-types'

const ErrorAlert = ({ message, dismiss }) => {
  const [visible, setVisible] = useState(true)

  const onDismiss = () => {
    setVisible(false)
    dismiss()
  }

  return message ? (
    <Alert color="danger" isOpen={visible} toggle={onDismiss}>
      {message}
    </Alert>
  ) : null
}

ErrorAlert.propTypes = {
  message: PropTypes.string,
  dismiss: PropTypes.func.isRequired,
}

export default ErrorAlert
