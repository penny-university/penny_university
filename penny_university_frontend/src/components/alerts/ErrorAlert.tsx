import React, { useState } from 'react'
import { Alert } from 'reactstrap'

const ErrorAlert = ({ message, dismiss }: { message: string | null, dismiss: () => void }) => {
  const [visible, setVisible] = useState(true)

  const onDismiss = () => {
    setVisible(false)
    dismiss()
  }

  return message ? (
    <Alert color="danger" isOpen={visible} toggle={onDismiss}>
      {message}
    </Alert>
  ) : null
}

export default ErrorAlert
