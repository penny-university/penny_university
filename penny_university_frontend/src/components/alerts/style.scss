.alert-container {
  bottom: 0;
  position: fixed;
  z-index: 10000;
}
