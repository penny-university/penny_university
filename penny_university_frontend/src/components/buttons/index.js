import HeartButton from './HeartButton'
import CreateButton from './CreateButton'
import EditButton from './EditButton'
import SaveButton from './SaveButton'
import CancelButton from './CancelButton'

export {
  HeartButton, CreateButton, EditButton, SaveButton, CancelButton,
}
