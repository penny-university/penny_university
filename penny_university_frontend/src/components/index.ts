import Modal from './modal'
import { Input } from './fields'

export {
  Modal,
  Input,
}
