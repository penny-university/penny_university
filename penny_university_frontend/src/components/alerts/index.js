import ErrorAlert from './ErrorAlert'

export { ErrorAlert }
