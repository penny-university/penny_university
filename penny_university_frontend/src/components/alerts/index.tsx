import React from 'react'
import { Alert } from 'reactstrap'
import { AnyAction } from 'redux'
import { ThunkDispatch } from 'redux-thunk'
import { connect } from 'react-redux'
import { ChatActions } from '../../actions'
import { RootState } from '../../reducers'

require('./style.scss')

type AlertProps = {
  error: Object | undefined,
  dismiss: () => void,
}

export const ErrorAlert = ({ error, dismiss }: AlertProps) => {
  const onDismiss = () => {
    dismiss()
  }

  return error ? (
    <div className="alert-container">
      <Alert color="danger" isOpen toggle={onDismiss}>
        {Object.values(error).flat().map((v, i) => <p key={`ErrorMessage-${v}`} className="mb-0">{v}</p>)}
      </Alert>
    </div>
  ) : null
}

const mapStateToProps = (state: RootState) => ({
  error: state?.error?.body,
})

const mapDispatchToProps = (dispatch: ThunkDispatch<{}, {}, AnyAction>) => ({
  dismiss: () => dispatch({ type: ChatActions.CLEAR_ERROR_MESSAGE }),
})

export default connect(mapStateToProps, mapDispatchToProps)(ErrorAlert)
