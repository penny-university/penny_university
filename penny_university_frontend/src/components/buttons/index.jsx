import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import {
  faPlus, faTimes, faPen, faSave, faHeart,
} from '@fortawesome/free-solid-svg-icons'
import classNames from 'classnames/dedupe'
import { Button } from 'reactstrap'
import PropTypes from 'prop-types'

require('./style.scss')

const IconButton = ({
  className, size, onClick, title, detail, icon, color,
}) => {
  const text = title && detail ? `${title} ${detail}` : title || detail
  return (
    <Button size={size} color={color} onClick={onClick} className={classNames(className, 'edit-button')}>
      <FontAwesomeIcon icon={icon} />
      {text}
    </Button>
  )
}

IconButton.propTypes = {
  size: PropTypes.string,
  title: PropTypes.string,
  detail: PropTypes.string,
  onClick: PropTypes.func,
  className: PropTypes.string,
  icon: PropTypes.element.isRequired,
  color: PropTypes.string,
}

IconButton.defaultProps = {
  size: 'md',
  color: 'primary',
  detail: '',
  className: '',
  title: '',
  onClick: null,
}

const CancelButton = ({
  className, onClick, type: detail,
}) => (
  <IconButton color="secondary" className={className} onClick={onClick} detail={detail} title="Cancel" icon={faTimes} />
)

CancelButton.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  type: PropTypes.string.isRequired,
}

CancelButton.defaultProps = {
  className: '',
}

const CreateButton = ({
  className, type: detail, onClick,
}) => (
  <IconButton onClick={onClick} className={className} detail={detail} title="Add New" icon={faPlus} />
)

CreateButton.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  type: PropTypes.string.isRequired,
}

CreateButton.defaultProps = {
  className: '',
}

const EditButton = ({
  className, onClick, type: detail,
}) => (
  <IconButton detail={detail} title="Edit" className={className} onClick={onClick} icon={faPen} />
)

EditButton.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  type: PropTypes.string.isRequired,
}

EditButton.defaultProps = {
  className: '',
}

const HeartButton = ({ className, count: detail }) => (
  <IconButton color="red" className={className} detail={detail} icon={faHeart} />
)

HeartButton.propTypes = {
  className: PropTypes.string,
  count: PropTypes.number.isRequired,
}

HeartButton.defaultProps = {
  className: '',
}

const SaveButton = ({
  className, type: detail, onClick,
}) => (
  <IconButton className={className} onClick={onClick} icon={faSave} title="Save" detail={detail} />
)

SaveButton.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  type: PropTypes.string.isRequired,
}

SaveButton.defaultProps = {
  className: '',
}

export {
  HeartButton, CreateButton, EditButton, SaveButton, CancelButton,
}

export default IconButton
