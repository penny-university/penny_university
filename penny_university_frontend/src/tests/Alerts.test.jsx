import React from 'react'
import { shallow, mount } from 'enzyme'
import { Alert } from 'reactstrap'
import { ErrorAlert } from '../components/alerts'

describe('error alert component', () => {
  it('renders correctly', () => {
    const component = shallow(<ErrorAlert message="error" />)
    expect(component).toMatchSnapshot()
  })

  it('should display with error message', () => {
    const component = mount(<ErrorAlert message="error" />)
    expect(component.find(Alert).text()).toContain('error')
  })

  it('should not display without an error message', () => {
    const component = shallow(<ErrorAlert />)
    expect(component.find(Alert)).toEqual({})
  })

  it('should be dismissed', () => {
    const component = mount(<ErrorAlert message="error" dismiss={jest.fn()} />)
    component
      .find('button.close')
      .at(0)
      .simulate('click')

    expect(component.find(Alert)).toEqual({})
  })
})
