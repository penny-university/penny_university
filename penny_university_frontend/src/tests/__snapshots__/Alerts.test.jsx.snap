// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error alert component renders correctly 1`] = `
<Alert
  closeAriaLabel="Close"
  color="danger"
  fade={true}
  isOpen={true}
  tag="div"
  toggle={[Function]}
  transition={
    Object {
      "appear": true,
      "baseClass": "fade",
      "baseClassActive": "show",
      "enter": true,
      "exit": true,
      "in": true,
      "mountOnEnter": false,
      "onEnter": [Function],
      "onEntered": [Function],
      "onEntering": [Function],
      "onExit": [Function],
      "onExited": [Function],
      "onExiting": [Function],
      "tag": "div",
      "timeout": 150,
      "unmountOnExit": true,
    }
  }
>
  error
</Alert>
`;
